ompi/runtime/ompi_mpi_abort.c

```diff
@@ -133,7 +133,7 @@ ompi_mpi_abort(struct ompi_communicator_t* comm,
     if (ompi_rte_initialized) {
         host = ompi_process_info.nodename;
     } else {
-        gethostname(hostname, sizeof(hostname));
+        gethostname(hostname, MAXHOSTNAMELEN);
```

Comment:
Is there a reason to make changes like this?  I somewhat like the defensive coding style of using `sizeof(foo)` to indicate the length of an array.  In this case, it's obviously `MAXHOSTNAME`, but using `sizeof` protects you in case the array changes size someday (i.e., there's only 1 place to update instead of 2).

Indeed, I wouldn't mind seeing:

1. Changing the size of all hostname-like arrays to `MAXHOSTNAMELEN` (which you do already in this PR)
1. Changing all references to the length of the array (e.g., passing as the 2nd arg to `gethostname()`) use `sizeof(array_name)`.